pkg/pulumiyaml/run.go

```diff
@@ -60,6 +60,7 @@ func LoadFromCompiler(compiler string, workingDirectory string) (*ast.TemplateDe
 	cmd.Dir = workingDirectory
 	cmd.Stdout = &stdout
 	cmd.Stderr = &stderr
+	cmd.Env = env
```

Comment:
I don't think this is backwards compatible to do.  Currently the compiler would inherit the environment from the process, but now we're only passing `PULUMI_STACK` in.  I don't think that is quite right, and we probably want to merge the processes environment and the environment we create internally.

Reply:
Yup, this needs to get the existing environment.

Reply:
Good point! Updated the PR and now mergin the `os.Environ()`